static/js/services/live-list.js

```diff
@@ -330,8 +337,8 @@ angular.module('inboxServices').factory('LiveList',
     }
 
     function _update(listName, updatedItem) {
-      _remove(listName, updatedItem);
-      _insert(listName, updatedItem);
+      var removed = _remove(listName, updatedItem);
```

Comment:
It's not exactly new. I've just changed the existing remove function to return the element that was removed, and pass it in to the insert function.

In the case of reports this means I can write code which sets the state of the checkbox of the new `li` element to the same as the old checkbox. This is fired for example when the changes feed notices an element in the list has changed. In this case we want the selection to remain the same but the doc content to be updated.

This is what fixed the bug you noticed where selections could get out of sync. In your case I suspect selecting the item marked it as read which fired the change listener which updated the $scope.selected just fine, but the checkbox was cleared when the livelist updated.